app/src/main/kotlin/com/wire/android/di/CoreLogicModule.kt

```diff
@@ -35,7 +42,28 @@ class CoreLogicModule {
 @Module
 @InstallIn(ViewModelComponent::class)
 class UseCaseModule {
+
+    @CurrentSession
+    @ViewModelScoped
+    @Provides
+    fun currentSessionProvider(@KaliumCoreLogic coreLogic: CoreLogic): AuthSession {
+        return runBlocking {
+            return@runBlocking when(val result = coreLogic.getAuthenticationScope().session.currentSession.invoke()) {
+                is CurrentSessionResult.Success -> result.authSession
+                else -> {throw IllegalStateException("no current session where found")}
+            }
+        }
+    }
```

Comment:
🤔 Are the providers called in the MainThread? I couldn't find a straight answer in 30s of Googling 😛 

In any case, one small query on a small table for sessions is kinda tolerable.

But we might want to optimise/cache the sessions in memory (in Kalium) to avoid querying the DB unnecessarily like in here.

Maybe just add a TODO here so we don't forget that this is a point we can improve.